newIDE/app/src/MainFrame/index.js

```diff
@@ -518,6 +522,29 @@ class MainFrame extends React.Component<Props, State> {
     this.toolbar.setEditorToolbar(editorToolbar);
   };
 
+  setScenePreview = () => {
+    const editorTab = getCurrentTab(this.state.editorTabs);
+    this._setLayoutForPreview(editorTab.name);
+
+    this.setState(
+      {
+        isPreviewOverride: true,
+      },
+      () => this.updateToolbar()
+    );
+  };
```

Comment:
I strongly think you don't need this whole function that is messing with this.state.editorTabs.

Just one function that you wrote:

```
  _setLayoutForPreview = (name:string) => {
    this.setState(
      {
        previewFirstSceneName: name,
        isPreviewOverride: true,
      },
      () => {
        this.updateToolbar();
      }
    );
  };
```

You need to call this function to set the name of the scene to launch. Do that when the user click on a toolbar, in other words, by doing:

```
<SceneEditor
  ..
  setScenePreview={() => {
    console.log("oh oh, seems the user want to use this scene as the preview scene! Let's remember it");
    this._setLayoutForPreview(name);
  }}
  ..
/>
```